app/lib/actions/candlepin/owner/async_import.rb

```diff
@@ -0,0 +1,23 @@
+module Actions
+  module Candlepin
+    module Owner
+      # This task is for polling the candlepin manifest import job.
+      # Candlepin::AbstractAsyncTask uses Actions::Base::Polling
+      # (which is an alias for Dynflow::Polling) and defines the run method to initiate polling.
+      # Therefore, don't override the run method here or in any dependent tasks.
+      class AsyncImport < Candlepin::AbstractAsyncTask
+        def poll_external_task(task_id = external_task[:id])
+          Rails.logger.info "Polling candlepin task"
+          task = ::Katello::Resources::Candlepin::Job.get(task_id, :result_data => true)
+          Rails.logger.info task['state']
+          task
+        end
```

Comment:
what was the reason to override this 'poll_external_task' method from Candlepin::AbstractAsyncTask ?

Reply:
It was just for the logging.  It's removed now since `AbstractAsyncTask` handles it

Reply:
Actually now that I think of it, I think I wasn't aware at the time that `AbstractAsyncTask` provides an implementation of `poll_external_task`.  I had just implemented it this way because I discovered `Katello::Resources::Candlepin::Job.get` was available and I needed it for the return value here.  I'm glad I arrived at basically the same code that was already there :smile: